src/material-experimental/mdc-progress-spinner/progress-spinner.ts

```diff
@@ -0,0 +1,262 @@
+    '[attr.aria-valuenow]': 'mode === "determinate" ? value : null',
+    '[attr.mode]': 'mode',
+  },
+  inputs: ['color'],
+  templateUrl: 'progress-spinner.html',
+  styleUrls: ['progress-spinner.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+})
+export class MatProgressSpinner extends _MatProgressSpinnerMixinBase implements AfterViewInit,
+  OnDestroy,
+  CanColor {
+  static ngAcceptInputType_diameter: NumberInput;
+  static ngAcceptInputType_strokeWidth: NumberInput;
+  static ngAcceptInputType_value: NumberInput;
+
+  /** Whether the _mat-animation-noopable class should be applied, disabling animations.  */
+  _noopAnimations: boolean;
+  /** Implements all of the logic of the MDC circular progress. */
+  _foundation: MDCCircularProgressFoundation;
+  @ViewChild('spinnerRoot') _rootElement: ElementRef<HTMLElement>;
+  @ViewChild('determinateSpinner') _determinateCircle: ElementRef<HTMLElement>;
+
+  private _adapter: MDCCircularProgressAdapter = {
+    addClass: (className: string) => this._rootElement.nativeElement.classList.add(className),
+    hasClass: (className: string) => this._rootElement.nativeElement.classList.contains(className),
+    removeClass: (className: string) => this._rootElement.nativeElement.classList.remove(className),
+    removeAttribute: (name: string) => this._rootElement.nativeElement.removeAttribute(name),
+    setAttribute: (name: string, value: string) =>
+      this._rootElement.nativeElement.setAttribute(name, value),
+    getDeterminateCircleAttribute: (attributeName: string) =>
+      this._determinateCircle.nativeElement.getAttribute(attributeName),
+    setDeterminateCircleAttribute: (attributeName: string, value: string) =>
+      this._determinateCircle.nativeElement.setAttribute(attributeName, value),
+  };
+  constructor(public _elementRef: ElementRef<HTMLElement>,
+              @Optional() @Inject(ANIMATION_MODULE_TYPE) animationMode: string,
+              @Inject(MAT_PROGRESS_SPINNER_DEFAULT_OPTIONS)
+                defaults?: MatProgressSpinnerDefaultOptions) {
+    super(_elementRef);
+    this._noopAnimations = animationMode === 'NoopAnimations' &&
+      (!!defaults && !defaults._forceAnimations);
+
+    if (defaults) {
+      if (defaults.diameter) {
+        this.diameter = defaults.diameter;
+      }
+
+      if (defaults.strokeWidth) {
+        this.strokeWidth = defaults.strokeWidth;
+      }
+    }
+  }
+
+  private _mode: ProgressSpinnerMode = 'determinate';
+
+  /**
+   * Mode of the progress bar.
+   *
+   * Input must be one of these values: determinate, indeterminate, buffer, query, defaults to
+   * 'determinate'.
+   * Mirrored to mode attribute.
+   */
+  @Input()
+  get mode(): ProgressSpinnerMode {
+    return this._mode;
+  }
+
+  set mode(value: ProgressSpinnerMode) {
+    // Note that we don't technically need a getter and a setter here,
+    // but we use it to match the behavior of the existing mat-progress-bar.
```

Comment:
```suggestion
    // but we use it to match the behavior of the existing mat-progress-spinner.
```
Is this comment accurate? It does seem like the setter is necessary to call `_syncFoundation`